core/src/main/java/com/sequenceiq/distrox/v1/distrox/StackUpgradeOperations.java

```diff
@@ -105,10 +105,11 @@ public UpgradeV4Response checkForClusterUpgrade(String accountId, @NotNull Stack
         if (replaceVms) {
             StackInstanceCount stackInstanceCount = instanceMetaDataService.countByStackId(stack.getId());
             Integer upgradeNodeCountLimit = limitConfiguration.getUpgradeNodeCountLimit();
-            if (stackInstanceCount.getInstanceCount() > upgradeNodeCountLimit) {
+            LOGGER.debug("Instance count: {} and limit: [{}]", stackInstanceCount == null ? "null" : stackInstanceCount.asString(), upgradeNodeCountLimit);
+            if (stackInstanceCount != null && stackInstanceCount.getInstanceCount() > upgradeNodeCountLimit) {
```

Comment:
what will happen when the stackInstanceCount is null? Do we enable the upgrade?

Reply:
yep, thats the idea, but as I wrote in the commit message nothing should be null there